modules/criteoIdSystem.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * This module adds Criteo Real Time User Sync to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/criteoIdSystem
+ * @requires module:modules/userId
+ */
+
+import * as utils from '../src/utils'
+import * as ajax from '../src/ajax'
+import { getRefererInfo } from '../src/refererDetection'
+import { submodule } from '../src/hook';
+
+const bididStorageKey = 'cto_bidid';
+const bundleStorageKey = 'cto_bundle';
+const cookieWriteableKey = 'cto_test_cookie';
+const cookiesMaxAge = 13 * 30 * 24 * 60 * 60 * 1000;
+
+const pastDateString = new Date(0).toString();
+const expirationString = new Date(utils.timestamp() + cookiesMaxAge).toString();
+
+function areCookiesWriteable() {
```

Comment:
cookiesAreEnabled function uses the window.navigator.cookieEnabled flag which returns true on Safari even on domains considered as trackers by ITP, which we would like to avoid. That's why we only rely on the read/write check.